TBB Build error on Clang 21.1.0-rc1

See https://jenkins.flatironinstitute.org/blue/organizations/jenkins/Stan%2FBleedingEdgeCompilersMonthly/detail/BleedingEdgeCompilersMonthly/272/pipeline/208

```
In file included from ./stan/math.hpp:19:
In file included from ./stan/math/rev.hpp:11:
In file included from ./stan/math/rev/constraint.hpp:6:
In file included from ./stan/math/rev/constraint/cholesky_corr_constrain.hpp:4:
In file included from ./stan/math/rev/core.hpp:14:
In file included from ./stan/math/rev/core/init_chainablestack.hpp:6:
In file included from lib/tbb_2020.3/include/tbb/task_scheduler_observer.h:25:
In file included from lib/tbb_2020.3/include/tbb/task_arena.h:23:
lib/tbb_2020.3/include/tbb/task.h:590:28: error: in-class initializer for static data member is not a constant expression
  590 |     static const kind_type binding_completed = kind_type(bound+1);
      |                            ^                   ~~~~~~~~~~~~~~~~~~
lib/tbb_2020.3/include/tbb/task.h:590:48: note: integer value 2 is outside the valid range of values [0, 1] for the enumeration type 'kind_type'
  590 |     static const kind_type binding_completed = kind_type(bound+1);
```

Presumably this is a small patch to fix, I'll try to see what other projects or upstream have changed.

cc @andrjohns @SteveBronder

--- a/lib/tbb_2020.3/STAN_CHANGES
+++ b/lib/tbb_2020.3/STAN_CHANGES
@@ -14,4 +14,4 @@ This file documents changes done for the stan-math project
     - L84 Wrapped the use of `-flifetime-dse` flag in conditional on non-WINARM64
     - L101 Wrapped the use of `-msse` in conditional on non-WINARM64
 
-- Add `constexpr` to lines 589-593 of `task.h` to fix a build failure in Clang 21.1
+- Added variants to the `kind_type` enum to prevent creation of out-of-range values

--- a/lib/tbb_2020.3/STAN_CHANGES
+++ b/lib/tbb_2020.3/STAN_CHANGES
@@ -13,4 +13,5 @@ This file documents changes done for the stan-math project
   - build/windows.gcc.inc
     - L84 Wrapped the use of `-flifetime-dse` flag in conditional on non-WINARM64
     - L101 Wrapped the use of `-msse` in conditional on non-WINARM64
-    
+
+- Add `constexpr` to lines 589-593 of `task.h` to fix a build failure in Clang 21.1

--- a/lib/tbb_2020.3/include/tbb/task.h
+++ b/lib/tbb_2020.3/include/tbb/task.h
@@ -376,7 +376,10 @@ private:
 public:
     enum kind_type {
         isolated,
-        bound
+        bound,
+        kind_complete,
+        kind_detached,
+        kind_dying
     };
 
     enum traits_type {
@@ -586,10 +589,10 @@ private:
     friend class task;
     friend class internal::allocate_root_with_context_proxy;
 
-    static constexpr kind_type binding_required = bound;
-    static constexpr kind_type binding_completed = kind_type(bound+1);
-    static constexpr kind_type detached = kind_type(binding_completed+1);
-    static constexpr kind_type dying = kind_type(detached+1);
+    static const kind_type binding_required = bound;
+    static const kind_type binding_completed = kind_type(bound+1);
+    static const kind_type detached = kind_type(binding_completed+1);
+    static const kind_type dying = kind_type(detached+1);
 
     //! Propagates any state change detected to *this, and as an optimisation possibly also upward along the heritage line.
     template <typename T>

--- a/lib/tbb_2020.3/include/tbb/task.h
+++ b/lib/tbb_2020.3/include/tbb/task.h
@@ -586,10 +586,10 @@ private:
     friend class task;
     friend class internal::allocate_root_with_context_proxy;
 
-    static const kind_type binding_required = bound;
-    static const kind_type binding_completed = kind_type(bound+1);
-    static const kind_type detached = kind_type(binding_completed+1);
-    static const kind_type dying = kind_type(detached+1);
+    static constexpr kind_type binding_required = bound;
+    static constexpr kind_type binding_completed = kind_type(bound+1);
+    static constexpr kind_type detached = kind_type(binding_completed+1);
+    static constexpr kind_type dying = kind_type(detached+1);
 
     //! Propagates any state change detected to *this, and as an optimisation possibly also upward along the heritage line.
     template <typename T>

